app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -145,6 +145,10 @@ def test_https_required
 
     def test_connection(options = {})
       super
+      connect(options)
+    end
+
+    def connect(options = {})
       if errors[:url].empty? && errors[:username].empty? && errors[:password].empty?
```

Comment:
Can we refactor this long `if` condition into a method, just to make code climate (and our eyes) happier?

We could also refactor this method to use quick return:

``` ruby
def connect(options = {})
  return unless connection_properties_valid?
  update_public_key options
  # ...
end
```